scenario evolution

cyrilphoenix71 opened the pull request:
fix #1570
fix #1558
fix #1554
fix #1582
- Ajout copier/couper/coller/remplacer des blocs scenario
- Ajout copier/couper/coller/remplacer des actions scenario
- ajout annuler/refaire ds prog scenario
- Ajout memoire temporaire (M+ M- Mreset) comme les calculatrice dans prg scenario
- Amelioration de l'autocompletion condition et action
- ajout de pleins d'action/conditions d'autocompletion 
- Ajout deplacement UP ou Down d'un bloc ou action
- Optimization affichage liste plugins actif/inactif

Pour tester
- verifier les issues
- tester a fond les scenarios

vinceg77 reviewed (commented):
- Page Paramètres : Pas de Listbox pour le champs Groupe (comme dans Jeedom certes) possibilité d'en ajouter une ?

- Autocomplétion : Fonctionne pour les champs 'Action' mais pas pour les champs 'Condition :' une liste de commande mais ne remplit pas la suite des caractères saisis. Du coup lorsque l'on clique dans la listebox d'autocomplétion, l’auto complétion vient s'ajouter aux caractères déjà saisis
![Autocompletion](https://user-images.githubusercontent.com/51484329/70376322-233f3880-1908-11ea-94cf-d2876c39c483.gif)

- Lors de l'utilisation des flèches de déplacement, lorsque l'on cliques sur une des fèches, le bloc sélectionné perd la sélection. Il faut donc le resélectionner à chaque fois afin de le remonter/redescendre à nouveau, ce qui n'est pas très pratique lorsqu'il y a de nombreux étages !

Tout le reste est bon, les features ajoutées par cette PR sont excellentes !

iWils reviewed (approved):
PR validé après plusieurs tests, les issues sont corrigés.
Toutefois j'ai rencontrer un bug avec le pied de page et le menu déroulant de choix:
![image](https://user-images.githubusercontent.com/56263502/70397386-0bec7200-1a12-11ea-91bb-c2953272e9a7.png)

cyrilphoenix71 commented:
> PR validé après plusieurs tests, les issues sont corrigés.
> Toutefois j'ai rencontrer un bug avec le pied de page et le menu déroulant de choix:
> ![image](https://user-images.githubusercontent.com/56263502/70397386-0bec7200-1a12-11ea-91bb-c2953272e9a7.png)

c corrigé

iWils reviewed (approved):
PR Approuvé

vinceg77 commented:
Retesté. tout est OK pour moi sauf : 
Si on copie une action et que l'on fait ensuite coller un bloc (action certes illogique mais qui peut se produire) : 
![CopieColle](https://user-images.githubusercontent.com/51484329/70617379-c8ac2200-1c10-11ea-9b91-03a8925c739c.gif)

Par contre si on fait l'inverse (copier un bloc et coller une action) ca colle l'action quand même (comportement acceptable)

Sylvaner reviewed (changes requested):
La PR a un champ trop large par rapport à son intitulé.
L'appel de la commande health à chaque page d'administration, au secours

cyrilphoenix71 committed:
scenario evolution

Déplacement des fichiers template

fix lazy

fix token
Copy/Paste scenario blocs
undo/redo


fix


missing expression action


optimize undo/redo


better autocompletion


rescue clean


mozilla scrollbar


gen-asset clean cache


better profil save


scenario links
better bloc dropdown


fix typeuse


fix


fix


Correction sonarcloud


fix #1582 + group list


fix z-index


optimize bloc move


fix condition autocompletion


oubli


fix


fix


fix


oubli

vinceg77 reviewed (approved):
Tout est désormais corrigé amélioré, optimisé et fonctionnel ! Gros boulot de @cyrilphoenix71, grosse PR !
👍
